test/GoogleAnalyticsEventProcessor/constructor_test.js

```diff
@@ -0,0 +1,89 @@
+goog.module('measurementLibrary.eventProcessor.testing.GoogleAnalyticsEventProcessor.processEvent');
+goog.setTestOnly();
+
+const GoogleAnalyticsEventProcessor = goog.require('measurementLibrary.eventProcessor.GoogleAnalyticsEventProcessor');
+
+describe('The `constructor` of GoogleAnalyticsEventProcessor', () => {
+  let eventProcessor;
+
+  const defaultAutomaticParams = {
+    'page_path': true,
+    'page_location': true,
+    'page_title': true,
+    'user_id': true,
+    'client_id': true,
+  };
+
+  it('does not have a default arguments for ' +
+      '`api_secret` or `measurement_id`', () => {
+    eventProcessor = new GoogleAnalyticsEventProcessor();
+
+    expect(eventProcessor.apiSecret_).toBeUndefined();
+    expect(eventProcessor.measurementId_).toBeUndefined();
+  });
+
+  it('has the correct default arguments for `measurement_url`, ' +
+  '`client_id_expires`, and `automatic_params`', () => {
+    eventProcessor = new GoogleAnalyticsEventProcessor();
+
+    expect(eventProcessor.measurementUrl_).toBe('https://www.google-analytics.com/mp/collect');
+    expect(eventProcessor.clientIdExpires_).toBePositiveInfinity();
+
+    for (const key in defaultAutomaticParams) {
+      expect(eventProcessor.automaticParams_.hasOwnProperty(key)).toBeTrue();
+
+      if (eventProcessor.automaticParams_.hasOwnProperty(key)) {
+        expect(eventProcessor.automaticParams_[key])
+            .toBe(defaultAutomaticParams[key]);
+      }
+    }
```

Comment:
I think just expect(automaticParams_).toEqual(defaultAutomaticaParams) is both cleaner and a stronger assertion.

I don't think any of the expectations in this loop would fail if defaultAutomaticParams had a key that was not in the eventProcessor.automaticParams_, but it probably should.

Reply:
I definitely forgot there was an toEqual() matcher 🤦 Updated!